### PR TITLE
Update ci.yaml to add documentation for archive properties

### PR DESCRIPTION
**Pull request by ricardoamador**
Updating CI.YAML.md to include definitions of some of the cache properties used in conjunction with the archive recipe. Though this could be extended.

*List which issues are fixed by this PR. You must list at least one issue.*
Part of https://github.com/flutter/flutter/issues/141688

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: CI_YAML.md [hunk 1]
```diff
@@ -188,6 +188,49 @@ Example
 presubmit_max_attempts: "3"
 ```
 
+#### Engine V2 specific
+
+##### archive
```
[the remark this answers is not in the document]
Comment (ricardoamador) [reply CI_YAML.md:193]: I think we will need to add this. I looked through the initial Engine_v2 docs, the design spec and the language spec and it appears that the cache was not included in the initial design. The original engine_v2.py, builder.py, tester.py came some time before that recipe. I will open a new issue to add this to the README.md in engine_v2.

[the remark this answers is not in the document]
Comment (ricardoamador) [reply CI_YAML.md:193]: Created https://github.com/flutter/flutter/issues/141883 for tracking. We will get to this next week during triage.

**Review by keyonghan — APPROVED** [review]
LGTM

File: CI_YAML.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (godofredoc) [reply CI_YAML.md:193]: The second layer cache was added as a fix to slow checkouts. I don't remember writing documents for that part.

[the remark this answers is not in the document]
Comment (godofredoc) [reply CI_YAML.md:193]: Taking a second look it seems that this is related to the swarming caches rather that second layer cache (engine v2). I'll take a look at the infra configurations to double check.

**Pull request closed without merging**